src/Hazelcast.Net/DistributedObjects/Impl/HSet.Removing.cs

```diff
@@ -25,30 +25,30 @@ public override async Task<bool> RemoveAsync(T item)
         {
             var itemData = ToSafeData(item);
             var requestMessage = SetRemoveCodec.EncodeRequest(Name, itemData);
-            var responseMessage = await Cluster.Messaging.SendAsync(requestMessage).CAF();
+            var responseMessage = await Cluster.Messaging.SendToKeyPartitionOwnerAsync(requestMessage, PartitionKeyData).CAF();
```

Comment:
For fixed partition types, List, Set, Queue etc. Invocation's use a fixed value derived from name.
So using `SendToPartitionOwnerAsync` and caching a `PartitionId` instead of `PartitionKeyData`  could be better to avoid partitionId calculation on hot path.

Reply:
Implemented in new commits below. Now ensuring it build in Jenkins. Want to review again?